### ISSUE TITLE
max player/server port/server password aren't configurable in the same manner

### ISSUE THREAD

**Issue by OnsenManju** [NyxStudios/TShock#1375]
As mentioned [here](https://tshock.co/xf/index.php?threads/quick-questions-thread-ask-your-how-do-i-and-where-is-questions-here.860/page-61#post-47684), these 3 settings are allowed to set at startup (before loading world) and ``config.json`` though, these values won't be reflected in the same manner.

(a) **max players** : will use config value even if you input desired value at startup
(b) **server port** : will use input value at startup, or use config value when ``-world`` option is used (but ``-port`` option overrides it)
(c) **server password** : won't use any value no matter what you input or set in config

(b) makes sense, but (a) is slightly confusing. (c) looks like a bug.

**Comment by nicatronTg** [comment]
It's worth noting that the precedence we should follow is:

1. Command line arguments take priority (because those are typically either set at runtime or before the server boots by a GSP).
2. Interactive options take priority next, because a user at a console should be able to modify their config on the fly if they're booting their server. However, if command line arguments are set, those are respected above all else.
3. Terraria config options should be taken into account next.
4. TShock config options should be taken into account last.

**Comment by hakusaro** [comment]
@NyxStudios/tshock thoughts on this?

**Comment by WhiteXZ** [comment]
I agree with your earlier comment

**Comment by mjevans** [comment]
I agree with Hakusaro's order of preference, but feel that an important 5th item exists that is merely implied instead of made an explicit part of the specification.

5. (well, always last) Default settings

**Comment by QuiCM** [comment]
We should probably do this

**Comment by Enerdy** [comment]
Is this still relevant?

**Comment by hakusaro** [comment]
Yes
